keycloak/keycloak_client.go

```diff
@@ -43,13 +43,13 @@ func NewKeycloakClient(baseUrl, clientId, clientSecret, realm, username, passwor
 	}
 	clientCredentials := &ClientCredentials{
 		ClientId: clientId,
+		ClientSecret: clientSecret,
```

Comment:
The zero value for a string is the empty string, so this shouldn't affect public clients.